common/fe-ste-models.go

```diff
@@ -395,6 +395,8 @@ func (BlobType) PageBlob() BlobType { return BlobType(2) }
 
 func (BlobType) AppendBlob() BlobType { return BlobType(3) }
 
+func (BlobType) Detect() BlobType { return BlobType(4) }
```

Comment:
this is replacing None as the default value, is None ever used? If not, it can be removed, right?

Reply:
If we remove None, then should we make Detect have the value zero?  (Given that zero-like values are kind of special in Go, as the default for new variables.  That's one thing that's nice about None, in that it's name implies zero-like-ness. But I agree that from a user's viewpoint its redundant)

Reply:
Indeed, the default value can be straight up `Detect`.